mods/taskbar-start-button-corner-fix.wh.cpp

```diff
@@ -0,0 +1,434 @@
+// @compilerOptions -lole32 -loleaut32 -luser32
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Start Menu Corner Click Fix
+
+Fixes an issue on Windows 11 where clicking in the very corner of the taskbar doesn't open the Start menu
+on left-aligned taskbars when the start menu button is "in a sticky corner".
+
+## The Problem
+
+Windows 11's taskbar has an InputSite window (Windows.UI.Input.InputSite.WindowClass)
+that intercepts mouse clicks before they reach the Start button.
+
+This seems to be caused by the "sticky corners" feature, and thus only affects multi-monitor setups.
+
+## The Solution
+
+This mod hooks `CreateWindowInBand` to detect when the InputSite window is created,
+then installs a thread-specific mouse hook on its thread. This hook intercepts mouse
+clicks at the corner region and invokes the Start menu via UI Automation when the
+real button doesn't handle the click.
+
+## Requirements
+
+- Windows 11 64-bit (tested on 25H2)
+- Windhawk v1.4 or later
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- edgeThickness: 5
+  $name: Edge thickness
+  $description: Thickness of the L-shaped edge region in pixels
+- edgeLength: 5
+  $name: Edge length
+  $description: How far the horizontal arm of the L extends from the corner in pixels
+- debugLogging: false
+  $name: Enable debug logging
+  $description: Log click events for debugging
+*/
+// ==/WindhawkModSettings==
+
+#include <windhawk_utils.h>
+
+#include <windows.h>
+#include <objbase.h>
+#include <uiautomation.h>
+
+// Settings
+struct {
+    int edgeThickness;
+    int edgeLength;
+    bool debugLogging;
+} g_settings;
+
+// Global state
+HHOOK g_mouseHook = NULL;
+bool g_inputSiteHooked = false;
+
+// Mouse hook state (only accessed from InputSite thread)
+bool g_mouseDownInCorner = false;
+bool g_menuOpenAtMouseDown = false;
+
+// UI Automation (initialized on InputSite thread)
+IUIAutomation* g_pAutomation = nullptr;
+HWND g_cleanupWindow = nullptr;
+const wchar_t* g_cleanupWindowClass = L"StartButtonFix_Cleanup";
+
+#define WM_CLEANUP_COM (WM_USER + 100)
+
+// Cleanup window procedure - runs on InputSite thread
+LRESULT CALLBACK CleanupWindowProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam) {
+    switch (uMsg) {
+        case WM_CLEANUP_COM:
+            if (g_pAutomation) {
+                g_pAutomation->Release();
+                g_pAutomation = nullptr;
+                CoUninitialize();
+                Wh_Log(L"UI Automation cleaned up on InputSite thread");
+            }
+            return 0;
+
+        case WM_CLOSE:
+            DestroyWindow(hWnd);
+            return 0;
+    }
+    return DefWindowProc(hWnd, uMsg, wParam, lParam);
+}
+
+// Initialize UI Automation (call from InputSite thread only)
+bool InitUIAutomation() {
+    if (g_pAutomation) return true;
+
+    HRESULT hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+    if (FAILED(hr) && hr != RPC_E_CHANGED_MODE && hr != S_FALSE) return false;

```

Comment:
* `S_FALSE` is success, so the check isn't needed.
* Calling `CoUninitialize` after getting `RPC_E_CHANGED_MODE` is incorrect, and generally it shouldn't happen.